processor/processor.go

```diff
@@ -1372,6 +1372,107 @@ func getDiffMetrics(inPU, pu string, inCountMetadataMap map[string]MetricMetadat
 	return diffMetrics
 }
 
+func generateFilteredAndAddedMetrics(inPU, pu string, messageIds []string, messageIdKeyMap map[string]string, inCountMetadataMap map[string]MetricMetadata, outputEvents []transformer.TransformerEvent, failedEvents []transformer.TransformerResponse) ([]*types.PUReportedMetric, []*types.PUReportedMetric) {
+	successMessageIdMap := make(map[string]int)
+	failedMessageIdMap := make(map[string]int)
+	for _, event := range outputEvents {
+		// handle transformBatch user transformation
+		// consider it as success if messageId is present in Metadata.MessageIDs of output event
+		// but do not increment to not count it as newly added event
+		if event.Metadata.MessageID == "" {
```

Comment:
@Sidddddarth is introducing a utility method in #3852. Wouldn't this simplify code?

```go
func (m Metadata) GetMessagesIDs() []string {
	if len(m.MessageIDs) > 0 {
		return m.MessageIDs
	}
	return []string{m.MessageID}
}
```

Reply:
Using empty messageID as proxy that output was from transformBatch transformation. Can be done similarly by checking length of GetMessagesIDs() but felt former was more explicit as that is the API contract between server - transformer